### PR TITLE
95 isAdmin status

### PR DESCRIPTION
**Pull request by ShevchenkoKirilo**
Now we have admin flag in database. Only admin users can see "Create lesson" button and save lessons to database

### CODE REVIEW

**Review by ishootdice — APPROVED** [review]

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs [hunk 1]
```diff
@@ -54,95 +54,117 @@ public async Task<UserDto> CreateUser(UserRegisterDto userDto)
...
     }
 
+    public async Task<bool> GetAdminStatusAsync()
+    {
+        var user = await _context.Users.FirstOrDefaultAsync(u => u.Id == _authService.UserId);
+        if (user is not null)
+        {
+            return user.IsAdmin;
+        }
+        return false;
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs:64]: You can get IsAdmin in SQL request, not to load all user

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/UserController.cs [hunk 2]
```diff
@@ -57,5 +59,15 @@ public async Task<ActionResult<List<UserShortInfoDto>>> GetSuitableUsers([FromBo
             return await _userService.GetFilteredUsers(userFilter);
         }
 
+        [HttpPut("makeAdmin/{userId}")]
+        public async Task<ActionResult<UserDto>> MakeAdminAsync(int userId, string secret)
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/UserController.cs:63]: Can you add secret to query?

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/UserController.cs [hunk 3]
```diff
@@ -57,5 +59,15 @@ public async Task<ActionResult<List<UserShortInfoDto>>> GetSuitableUsers([FromBo
...
+        {
+            var settingsSecret = _configuration.GetSection("MakeAdminSecret").Value;
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/UserController.cs:65]: MakeAdminSecret -> adminSecret

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts [hunk 4]
```diff
@@ -41,6 +43,12 @@ export class LessonsPageComponent implements OnInit, OnChanges {
         this.getLessons();
 
         this.todayDate = moment().format('DD MMMM YYYY, dddd');
+
+        const user = localStorage.getItem('user');
+
+        if (user) {
+            this.userIsAdmin = JSON.parse(user).isAdmin;
+        }
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts:51]: Load it from service, please, don't use localStorage in components

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/UserController.cs (hunk 2 above)
Comment (ShevchenkoKirilo) [reply backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Controllers/UserController.cs:63]: Can you explain in more detail what should I change

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by ishootdice — APPROVED** [review]

**Pull request merged**